api/src/pcapi/routes/adage/v1/prebooking.py

```diff
@@ -101,9 +100,33 @@ def get_all_bookings_per_year(
     educational_year_id: str,
     query: prebooking_serialization.GetAllBookingsPerYearQueryModel,
 ) -> prebooking_serialization.EducationalBookingsPerYearResponse:
-    educational_bookings = get_paginated_collective_bookings_for_educational_year(
+    educational_bookings = educational_repository.get_paginated_collective_bookings_for_educational_year(
         educational_year_id,
         query.page,
         query.per_page,
     )
     return prebooking_serialization.get_collective_bookings_per_year_response(educational_bookings)
+
+
+@blueprint.adage_v1.route("/prebookings/move", methods=["POST"])
+@spectree_serialize(
+    api=blueprint.api,
+    on_success_status=204,
+    on_error_statuses=[404, 422],
+    tags=("merge institution"),
+)
+@adage_api_key_required
+def merge_institution_prebooking(body: prebooking_serialization.BookingsInstitutionQueryModel) -> None:
+    try:
+        bookings = educational_repository.find_collective_bookings_by_institution(body.source)
+    except exceptions.EducationalInstitutionSourceNotFound:
+        raise ApiErrors(
+            {"code": "Source institution not found or no booking linked to that source institution"}, status_code=404
+        )
+    for booking in bookings:
+        booking.educationalInstitution = educational_repository.find_educational_institution_by_uai_code(
+            body.destination
+        )
+        if not booking.educationalInstitution:
+            raise ApiErrors({"code": "New institution not found"}, status_code=404)
```

Comment:
Ici on fait une requête SQL (dans `find_educational_institution_by_uai_code()`) par réservation. Il serait préférable de récupérer l'institution hors de la boucle.

```suggestion
    destination = educational_repository.find_educational_institution_by_uai_code(body.destination)
    if not destination:
        return ApiErrors(..., 404)
    for booking in bookings:
        ...
```